app/app-controllers/tabBarController.js

```diff
@@ -100,7 +100,18 @@ function(
             this.viewPlugin.setContentView(this.tabBar.tabViews[tabId]);
             this.activeTabId = tabId;
 
-            this.getActiveNavigationView().isActive = true;
+            var selectedTab = this.getActiveNavigationView();
+            selectedTab.isActive = true;
+
+            // TODO: Rethink isLoaded exposure
```

Comment:
There is a hole in the logic here :'(. Currently `.loaded` on a navigationView means that a page attempted to navigate and never ran into a `pageTimeout` or `noInternetConnection` error. But if it ever does run into an error, we want it to reload the page that it was going to rather than the root url.

I have an incoming commit for address this.